src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -244,7 +267,14 @@ func rowsToEvents(rows *sql.Rows) ([]gomatrixserverlib.Event, error) {
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, ev)
+		if order == rowOrderForward {
+			result = append(result, ev)
+		} else if order == rowOrderReverse {
+			result = append([]gomatrixserverlib.Event{ev}, result...)
```

Comment:
My main complaint is with the clarity of the rowOrdering options, but it's worth noting that "appending" to the front of a slice is less efficient than appending to the end of the slice.